### PR TITLE
Ignore failing tests

### LINKED ISSUE #3483: failing test - tests/core/htmldataprocessor/htmldataprocessor
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. run `tests/core/htmldataprocessor/htmldataprocessor` on iOS13 with Safari

### Expected result

passing test

### Actual result

failing test

## Other details

IMG_0005

* Browser: Safari
* OS: iOS 13 (iPhone 11)
* CKEditor version: adcefdd71 (4.13.0)
* Installed CKEditor plugins: …

### LINKED ISSUE #4111: Ignore failing tests
## Type of report

Task

## Provide description of the task

Tests that are reported to be failing or unstable but would take too much time to fix should be ignored to reduce distraction during future testing.

### PR DESCRIPTION
**Pull request by Dumluregn**
## What is the purpose of this pull request?

Task

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- ignorancy

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

No changelog entry.

## What changes did you make?

Ignored failing tests according to the list from #3794. In general I was trying to only ignore particular cases, but sometimes it wasn't possible as the next tests started to fail as they depended on the outcome of the previous ones. Also sometimes it wasn't clearly indicated in the tickets which test cases are unstable, so it is possible we will need to ignore more than I did.

## Which issues does your PR resolve?

Closes #4111.
Closes #3483.

### CODE REVIEW

File: tests/plugins/pastetools/filter/parsestyles.js [hunk 1]
```diff
@@ -6,6 +6,11 @@
 ( function() {
 	'use strict';
 
+	// (#3713)
+	if ( CKEDITOR.env.edge && CKEDITOR.env.version < 80 ) {
+		bender.ignore();
+	}
+
```
Comment (f1ames) [inline tests/plugins/pastetools/filter/parsestyles.js:13]: Not sure if we would ever need to run this tests on older Edge versions (\`edgeHTML\`) :thinking: Maybe we should just leave it and also close #3713 as Edge migrated to Chromium.

File: tests/core/editor/detaching.js [hunk 2]
```diff
@@ -305,6 +305,11 @@
 	// @returns {Function}
 	function getAfterIframeLoadTestCase( options ) {
 		return function() {
+			// (#3714)
+			if ( CKEDITOR.env.edge && CKEDITOR.env.version < 80 ) {
+				assert.ignore();
+			}
+
```
Comment (f1ames) [inline tests/core/editor/detaching.js:312]: See my other comment regarding edgeHtml.

File: tests/plugins/easyimage/widget.js [hunk 3]
```diff
@@ -84,10 +84,15 @@
...
-				bender.tools.ignoreUnsupportedEnvironment( 'easyimage' );
+			bender.tools.ignoreUnsupportedEnvironment( 'easyimage' );
 		},
 
 		'test upcasting image widget (figure)': function( editor, bot ) {
+			// (#3715)
+			if ( CKEDITOR.env.edge && CKEDITOR.env.version < 80 ) {
+				assert.ignore();
+			}
+
```
Comment (f1ames) [inline tests/plugins/easyimage/widget.js:95]: See my other comment regarding edgeHtml.

File: tests/core/htmldataprocessor/htmldataprocessor.js [hunk 4]
```diff
@@ -30,6 +30,11 @@
 			output = output ? output.replace( /%xss%/g, 'testXss()' ) : input;
 
 		tcs[ name ] = function() {
+			// (#3483)
```
Comment (f1ames) [inline tests/core/htmldataprocessor/htmldataprocessor.js:33]: I wonder if we shouldn't just try to fix this test (#3483) as it seems like the matter of porper HTML normalization before assertion?

Comment (Dumluregn) [reply tests/core/htmldataprocessor/htmldataprocessor.js:33]: It's caused by changes in Safari engine - browsers were resigning from removing preceding spaces from attribute value (see #1070 and #1291 for reference), Safari was the only one left on that field, but it also adapted. I changed the test accordingly (actually removing browser sniffing instead of doing it twice, so it's a good change :joy: ).

**Comment by f1ames** [comment]
Rebased onto latest `master`.
